### PR TITLE
[SPARK-38446][Core] Fix deadlock between ExecutorClassLoader and FileDownloadCallback caused by Log4j

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

While `log4j.ignoreTCL/log4j2.ignoreTCL` is false, which is the default, it uses the context ClassLoader for the current Thread, see `org.apache.logging.log4j.util.LoaderUtil.loadClass`. While ExecutorClassLoader try to loadClass through remotely though the FileDownload, if error occurs, we will long on debug level, and `log4j...LoaderUtil` will be blocked by ExecutorClassLoader acquired classloading lock.

Fortunately, it only happens when ThresholdFilter's level is `debug`.

or we can set `log4j.ignoreTCL/log4j2.ignoreTCL` to true, but I don't know what else it will cause.

So in this PR, I simply remove the debug log which cause this deadlock

### Why are the changes needed?

fix deadlock

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

https://github.com/apache/incubator-kyuubi/pull/2046#discussion_r821414439, with a ut in kyuubi project, resolved(https://github.com/apache/incubator-kyuubi/actions/runs/1950222737)

### Additional Resources

[ut.jstack.txt](https://github.com/apache/spark/files/8206457/ut.jstack.txt)

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @HyukjinKwon @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
cc @viirya

File: core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala:454]: So, the deadlock happens when we use DEBUG log level?

Comment (yaooqinn) [reply core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala:454]: the log4j will loadClass again with `ExecutorClassLoader`, if the loglevel does not satisfy, it is skipped

Comment (yaooqinn) [reply core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala:454]: Before the deadlock occurs, I see something very similar with https://github.com/apache/spark/pull/10337#issuecomment-165266841

**Comment by cloud-fan** [comment]
cc @Ngone51 @jiangxb1987

**Comment by viirya** [comment]
Hmm, after I read the description, I think shouldn't we fix it at `ExecutorClassLoader`? E.g., falling back to the default class loader as log4j2.ignoreTCL?

**Comment by yaooqinn** [comment]
thanks, @viirya,
Firstly we cannot fall back directly before ExecutorClassLoader tries remote classes via log4j2.ignoreTCL, it fails in the middle of fetching remote classes. Then we try to log a debug message where we shall skip via log4j2.ignoreTCL. As we cannot determine in ExecutorClassLoader whether the caller for loading classes is from Log4j. I don't know how we play the trick to ignore log4j2.ignoreTCL first then respect it later.

**Comment by dongjoon-hyun** [comment]
cc @MaxGekk since he is a release manager.

Although this is still unclear to me, this could be a regression at Apache Spark 3.3.0 release.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Since this is a removal of log, there is no functional regression . 
Let me merge this and I hope this can unblock Apache Kyuubi project and Apache Spark 3.3.0 release.

Thank you, @yaooqinn , @srowen , @viirya , @cloud-fan .

**Comment by yaooqinn** [comment]
thanks @dongjoon-hyun and all

**Pull request closed without merging**
